mesonbuild/compilers/compilers.py

```diff
@@ -1155,8 +1154,16 @@ def compile(self, code, extra_args=None, mode='link', want_output=False):
                 p.input_name = srcname
                 if want_output:
                     p.output_name = output
-                else:
-                    self.compiler_check_cache[key] = p
+                elif cdata is not None:
+                    # Remove all attributes except the following
+                    # This way the object can be serialized
+                    tokeep = ['args', 'commands', 'input_name', 'output_name',
+                              'pid', 'returncode', 'stdo', 'stde', 'text_mode']
+                    todel = [x for x in vars(p).keys() if x not in tokeep]
+                    for i in todel:
+                        delattr(p, i)
```

Comment:
I wonder if the output of Popen_safe should not always be copied to a new object with just the properties that are needed. That way this functions return would not depend on if cdata is None.

Reply:
When writing this, I found that some parts of the return value of `Popen_safe` save where accesed down the road. So changing this would requrie having a really clos looking at the entire compiers subfolder.

Reply:
I was thinking of always reducing the object to only contain tokeep. Or do you suspect that parts of the code that have caching disabled are using even more attributes?

Reply:
I knew when originally writing this, but to give a definite answer I would have to look through the entire compiler code again.

I only know that somewhere `compile` is called without coredata so that nothing is cached and that some other value than `returncode` is accessed.